bin/cli.js

```diff
@@ -29,17 +35,36 @@ function getParentBranches() {
 }
 
 function bump(type) {
-    const publishedVersions = JSON.stringify(execSync(`npm show . versions -json`, {encoding: 'utf8'}));
     let newVersion;
-    do {
-        newVersion = execSync(`npm version ${type}`, {encoding: 'utf8'}).trim().substr(1);
-    } while(publishedVersions.indexOf(newVersion) !== -1);
+
+    var fs = require('fs');
+    if (fs.existsSync('pom.xml')) {
+        const publishedVersion = JSON.stringify(execSync(`mvn -q -Dexec.executable="echo" -Dexec.args='\${project.version}' --non-recursive exec:exec`, {encoding: 'utf8'})).replace("\\n","");
+        console.log('Detected current version :', publishedVersion);
+
+        const parsedVersion = publishedVersion.match("([0-9]+){1}\.([0-9]+){1}\.([0-9]+){1}");
+        if (parsedVersion.length != 4) {
+            console.log('Did not match a SemVer valid version, please bump manually to a SemVer format.')
+            return;
+        }
+        
+        parsedVersion[VersionPosition[type]] =  (parseInt(parsedVersion[VersionPosition[type]], 10) + 1).toString()
+        newVersion = parsedVersion[VersionPosition[Version.MAJOR]] + "." + parsedVersion[VersionPosition[Version.MINOR]] + "." + parsedVersion[VersionPosition[Version.PATCH]];
+        console.log('Bumping to new version :', newVersion)
+
+        JSON.stringify(execSync(`mvn versions:set -DnewVersion=${newVersion}`, {encoding: 'utf8'})).replace("\\n","");
+    } else {
+        const publishedVersions = JSON.stringify(execSync(`npm show . versions -json`, {encoding: 'utf8'}));
+        do {
+            newVersion = execSync(`npm version ${type}`, {encoding: 'utf8'}).trim().substr(1);
+        } while(publishedVersions.indexOf(newVersion) !== -1);
```

Comment:
Here we do a loop to avoid publishing over an existing version, I don't know if you'll need it in maven

Reply:
You can't republish an existing version in maven, at least when you set it up correctly. I don't think we need to loop to make sure a version is published or not. This script IMO should only be to bump the correct number according to what branch was merged and the user is responsible for using it on the last published version. Don't you think? Did you have any problems with that in the past for npm?

Reply:
We had some issues with concurrent builds in Jenkins but it doesn't happens often